### ISSUE TITLE
MongoDb Replica Set Support

### ISSUE THREAD

**Issue by ndcollins** [pencilblue/pencilblue#738]
It does not appear that PencilBlue's MongoDB configuration supports connecting to a MongoDB ReplicaSet. When I try, even if the generated connection string is identical to what I use with MongoChef to successfully connect, PencilBlue gives me an authentication error.

**Comment by brianhyder** [comment]
@ndcollins Would you be willing to send us an email with your connection info and help you debug it?

**Comment by ndcollins** [comment]
Already sent it on Tuesday to your brian(at) email address with the subject "PencilBlue & Replica Sets"

Sent from my iPhone

On Sep 8, 2015, at 5:15 PM, Brian <notifications@github.com<mailto:notifications@github.com>> wrote:


@ndcollins<https://github.com/ndcollins> Would you be willing to send us an email with your connection info and help you debug it?

-

**Comment by brianhyder** [comment]
I took a look at the setup this evening.  The driver does not seem to like the connection string.  I was able to create a branch and get an experimental fix in: https://github.com/pencilblue/pencilblue/tree/issue/738

You will need to configure your config.js file's db section like the following:
```
"db": {
        "type":"mongo",
        "servers": [
          "server1:8781",
          "server2:8927"
        ],
        "name": "db_name",
        "authentication": {
            un: 'username',
            pw: 'password'
        },
        options: {
            replSet: {
                replicaSet: 'set-xyz'
            }
        }
    }
```

**Comment by brianhyder** [comment]
The driver seems to intermittently complete the authentication and other times fail.  If you get a chance can you see what results you are seeing?

**Comment by brianhyder** [comment]
@ndcollins I did a little more testing.  I was able to get it to connect to the replica set that you provided.  The fix was merged into the 0.5.0 branch.  The sample configuration shown below demonstrates how to connect PB to a replica set.

```
"db": {
        "type":"mongo",
        "servers": [
          "server1:8781",
          "server2:8927"
        ],
        "name": "db_name",
        "authentication": {
            un: 'username',
            pw: 'password'
        },
        options: {
            replSet: {
                replicaSet: 'set-xyz'
            }
        }
    }
```
